applications/StructuralMechanicsApplication/custom_elements/truss_element_3D2N.cpp

```diff
@@ -408,6 +410,75 @@ namespace Kratos
 		KRATOS_CATCH("")
 	}
 
+	void TrussElement3D2N::CalculateOnIntegrationPoints(
+		const Variable<array_1d<double, 3 > >& rVariable,
+		std::vector< array_1d<double, 3 > >& rOutput,
+		const ProcessInfo& rCurrentProcessInfo) 
+		{
+		if (rVariable == FORCE)
+				{
+					Vector truss_forces = ZeroVector(msDimension);
+					truss_forces[2] = 0.00;
+					truss_forces[1] = 0.00;
+
+					const double internal_strain_gl = this->CalculateGreenLagrangeStrain();
+
+					const double L0 = this->CalculateReferenceLength();
+					const double l = this->CalculateCurrentLength();
+					const double E = this->GetProperties()[YOUNG_MODULUS];
+					const double A = this->GetProperties()[CROSS_AREA];
+
+					double s_pre = 0.00;
+					if (this->GetProperties().Has(TRUSS_PRESTRESS_PK2)) {
+						s_pre = this->GetProperties()[TRUSS_PRESTRESS_PK2];
+					}
+
+					truss_forces[0] = ((E*internal_strain_gl + s_pre) * l * A) / L0;

```

Comment:
i see that you are using a linear displacement measure, l/l0. what kind of formulation are u using here?

also why do you need to do something different in the linear case?

Reply:
for linear strain meassurement only the change of length as a projection wrt the reference configuration is taken. For non-linear cases I need the actual change of length